sacc2/sacc.py

```diff
@@ -0,0 +1,871 @@
+        """
+        return unique_list(d.data_type for d in self.data)
+
+    def get_tracer(self, name):
+        """
+        Get the tracer object with the given name
+
+        Parameters
+        -----------
+        name: str
+            A string name of a tracer
+
+        Returns
+        -------
+        tracer: BaseTracer object
+            The object corresponding to the name.
+        """
+        return self.tracers[name]
+
+    def get_tracer_combinations(self, data_type=None):
+        """
+        Find all the tracer combinations (e.g. tomographic bin pairs)
+        for the given data type
+
+        Parameters
+        -----------
+        data_type: str
+            A string name of the data type to find
+
+        Returns
+        -------
+        combinations: list of tuples of strings
+            A list of all the tracer combinations found
+            in any data point.  No specific ordering.
+        """
+        indices = self.indices(data_type=data_type)
+        return unique_list(self.data[i].tracers for i in indices)
+
+    @property
+    def mean(self):
+        """
+        Get the vector of mean values for the entire data set.
+
+        Returns
+        -------
+        mean: array
+            numpy array with all the mean values in the data set
+        """
+        return np.array([d.value for d in self.data])
+
+    @mean.setter
+    def mean(self, mu):
+        """
+        Set the vector of mean values for the entire data set.
+
+        Parameters
+        -----------
+
+        mu: array
+            Replace the mean values of all the data points.
+        """
+        if not len(mu) == len(self.data):
+            raise ValueError("Tried to set mean with thing of length {}"
+                             " but data is length {}".format(len(mu), len(self.data)))
+        for m, d in zip(mu, self.data):
+            d.value = m
+
+    def _make_window_tables(self):
+        # Convert any window objects in the data set to tables,
+        # and record a mapping from those objects to table references
+        # This could easily be extended to other types
+        all_windows = unique_list(d.get_tag('window') for d in self.data)
+        window_ids = {w: id(w) for w in all_windows}
+        tables = BaseWindow.to_tables(all_windows)
+        return tables, window_ids
+
+    def save_fits(self, filename, overwrite=False):
+        """
+        Save this data set to a FITS format Sacc file.
+
+        Parameters
+        ----------
+
+        filename: str
+            Destination FITS file name
+
+        overwrite: bool
+            If False (the default), raise an error if the file already exists
+            If True, overwrite the file silently.
+
+        """
+
+        # Since we don't want to re-order the file as a side effect
+        # we first make a copy of ourself and re-order that.
+        S = self.copy()
+        S.to_canonical_order()
```

Comment:
It's not 100% necessary, but I thought it would be useful since then two files should look much more similar to each other if they have the same content.

Reply:
Right, I don't disagree. There are however applications where having a different ordering is more optimal (e.g. to fully account for the non-Gaussian likelihood of the C_ells on large scales a la https://arxiv.org/abs/0801.0554 it's better to have the bin indices be the fastest-growing ones, rather than ell). So could we make this reordering optional through a keyword argument that defaults to `True`?

Reply:
sounds good - will do.

Reply:
Actually I've just remembered the other reasoning here - we need to group together the data points which share the same data_type, since they will be saved in a single table together.  If we don't do this then the covariance will be in the wrong order.

You can imagine a user e.g. alternating adding EE and BB data points for the same ell, but when we save we need to re-group them.

Reply:
OK, I see. I didn't know these were being saved in different tables. I guess forget about this then.